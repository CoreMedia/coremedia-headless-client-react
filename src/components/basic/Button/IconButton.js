// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  color?: string,
  backgroundColor?: string,
};

const IconButton: ComponentType<Props> = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: ${props => props.color || 'inherit'};
  background-color: ${props => props.backgroundColor || 'transparent'};
  cursor: pointer;
`;

IconButton.displayName = 'IconButton';

IconButton.propTypes = {
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
};

export default IconButton;
