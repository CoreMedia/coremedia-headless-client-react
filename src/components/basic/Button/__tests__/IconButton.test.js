// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import IconButton from '../IconButton';

describe('IconButton Component', () => {
  it('should render correctly with default props', () => {
    const tree = renderer.create(<IconButton>Content</IconButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with given color', () => {
    const tree = renderer.create(<IconButton color="#fff">Content</IconButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with given backgroundColor', () => {
    const tree = renderer.create(<IconButton backgroundColor="#fff">Content</IconButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
