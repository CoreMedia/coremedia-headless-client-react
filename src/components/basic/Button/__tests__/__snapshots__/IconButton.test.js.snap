// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component should render correctly with default props 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
}

<button
  className="c0"
>
  Content
</button>
`;

exports[`IconButton Component should render correctly with given backgroundColor 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: inherit;
  background-color: #fff;
  cursor: pointer;
}

<button
  className="c0"
>
  Content
</button>
`;

exports[`IconButton Component should render correctly with given color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: #fff;
  background-color: transparent;
  cursor: pointer;
}

<button
  className="c0"
  color="#fff"
>
  Content
</button>
`;
